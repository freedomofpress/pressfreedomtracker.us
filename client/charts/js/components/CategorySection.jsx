import React from 'react'
import PropTypes from 'prop-types'

export default function CategorySection({
	label,
	count,
	isOpen,
	onClick,
	children,
}) {
	return (
		<div className="filters__group">
			<div onClick={() => onClick(label)}>
				{!isOpen && <span>&#9650;</span>}
				{isOpen && <span>&#9660;</span>}
				{label} ({count})
			</div>
			{isOpen && (
				<>
					{children}
				</>
			)}
		</div>
	)
}

CategorySection.propTypes = {
	label: PropTypes.string.isRequired,
	count: PropTypes.number.isRequired,
	isOpen: PropTypes.bool.isRequired,
	onClick: PropTypes.func.isRequired,
	children: PropTypes.node,
}
