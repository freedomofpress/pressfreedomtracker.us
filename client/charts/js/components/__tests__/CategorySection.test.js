import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import CategorySection from '../CategorySection'

test('renders CategorySection with mocked data', () => {
	expect(ReactTestRenderer.create(
		<CategorySection symbol="test" label="test" count={2} isOpen={false} onClick={() => {}}>
			This is a test
		</CategorySection>
	)).toMatchSnapshot();
});
