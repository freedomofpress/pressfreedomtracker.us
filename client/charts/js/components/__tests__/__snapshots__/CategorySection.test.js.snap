// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CategorySection with mocked data 1`] = `
<div
  className="category-checkbox"
>
  <input
    checked={false}
    className="category-checkbox--input"
    id="test"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="category-checkbox--label"
    htmlFor="test"
  >
    <span
      className="category category-test"
    >
      test
    </span>
    <span>
      2
    </span>
  </label>
</div>
`;
